src/climate_resilience/bias_correction.py

```diff
@@ -112,14 +112,15 @@ def read_observation_data(
         pd.DataFrame: Data frame of observational data for given site.
 
     """
-  for filename in os.listdir(obsdir  + "Site_" + var +"/"):
-    if site in filename: 
-      data = pd.read_csv(obsdir + "Site_" + var + "/" +filename)
-      data = pd.read_csv(obsdir + "Site_" + var + "/" +filename)
-      if "Date" in data.columns:
-        data = data.set_index('Date')
-      if "date" in data.columns:
-        data = data.set_index('date')
+    
+    for filename in os.listdir(obsdir  + "Site_" + var +"/"):
+      if site in filename: 
+        data = pd.read_csv(obsdir + "Site_" + var + "/" +filename)
+        data = pd.read_csv(obsdir + "Site_" + var + "/" +filename)
+        if "Date" in data.columns:
+          data = data.set_index('Date')
+        if "date" in data.columns:
+          data = data.set_index('date')
 
       return data
```

Comment:
The "return" command is inside the "for" loop.
This way, it'll always return in the first iteration and the next iteration will never run.
Can you please check and correct this?